.github/workflows/ghcr.yml

```
@@ -62,15 +62,16 @@ jobs:
         shell: python
         run: |
           import json
+          import os
           metadata = json.loads('${{ steps.metadata.outputs.json }}')
           version = metadata["labels"]["org.opencontainers.image.version"]
           if version == "main":
               # get first version that isn't main
               for tag in metadata["tags"]:
                   if ":main" not in tag:
                       version = tag.split(":")[1]
-
-          print(f"::set-output name=version::{version}")
+          with open(os.environ.get("GITHUB_STATE"), "a") as state:
```

```suggestion
          with open(os.environ.get("GITHUB_OUTPUT"), "a") as state:
```
Changing to `GITHUB_OUTPUT` should keep the rest of the flow the same. Curious if there is an advantage on using `GITHUB_STATE`?

---

You're right, @krysal, I mixed up the examples with `state` and `output` here: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/#examples